compiler/passes/docs.cpp

```diff
@@ -500,8 +519,23 @@ void printFunction(std::ofstream *file, FnSymbol *fn, bool method) {
     if (fn->isSecondaryMethod()) {
       if (fn->numFormals() > 1) {
         ArgSymbol *myTypeHolder = fn->getFormal(2);
-        if (myTypeHolder->hasFlag(FLAG_ARG_THIS))
-          *file << myTypeHolder->type->symbol->name << ".";
+        if (myTypeHolder->hasFlag(FLAG_ARG_THIS)) {
+          Expr *typeExpr = myTypeHolder->typeExpr;
+          if (BlockStmt *body = toBlockStmt(typeExpr)) {
+            if (UnresolvedSymExpr *typeName = toUnresolvedSymExpr(body->body.tail)) {
+              *file << typeName->unresolved << ".";
+            } else {
+              INT_ASSERT(false);
```

Comment:
It might make the code a bit more readable, and the assertions more useful if they ever do happen.